src/compiler.ts

```diff
@@ -4661,6 +4662,7 @@ export class Compiler extends DiagnosticEmitter {
           }
           this.currentType = leftType;
         }
+        this.currentFlow.inheritBranch(rightFlow);
```

Comment:
Similarly, if the LHS is always `false`, the parent flow would inherit the right flow unconditionally, while if it is always `true`, it would not inherit anything since it never executes.